File: Part 2 - Regression/Section 4 - Simple Linear Regression/combined_simple_regression.py
# -*- coding: utf-8 -*-
"""
Created on Mon Aug 20 22:03:40 2018

@author: Hrishika
"""

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, train_size = 0.8, random_state = 0)

#For simple regression the library handles feature scaling
#Import linear regression library
from sklearn.linear_model import LinearRegression
sim_reg = LinearRegression()

#Train the simple linear regressor on the training set
sim_reg.fit(X_train, y_train)

#Testing the regressor on the test set
y_predicted = sim_reg.predict(X_test)

#Plot the results test vs predicted
#Plot the training set
plt.scatter(X_train, y_train, color= 'red')
plt.plot(X_train, sim_reg.predict(X_train), color='blue')
plt.title('Salary vs Experience (Training set results)')
plt.xlabel('Experience in Years')
plt.ylabel('Salary')
plt.show()

#Plot the test set
plt.scatter(X_test, y_test, color= 'red')
plt.plot(X_train, sim_reg.predict(X_train), color='blue')
plt.title('Salary vs Experience (Test set results)')
plt.xlabel('Experience in Years')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 4 - Simple Linear Regression/combined_simple_regression.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('Salary_Data.csv')

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#fit the training set to the simple linear regression model
regressor <- lm(formula = Salary ~ YearsExperience,
                data = training_set)

summary(regressor)
#run summary on the regressor to see the various statistical parameters such as p value
# multiple r squared etc. These model parameters can give us important info about the model
#for the below example the variable of experience is marked with 3 stars which shows
# that experience as a variable is statistically significant. The very very small p value
# also supports this. This gives us an indication that the formula we specified Salary ~ Years
# is the right one


#Test the accuracy of the regressor on the test set
predicted_results <- predict(regressor, newdata = test_set)

#Plot the training set 
library(ggplot2)
p_train <- ggplot(data = training_set)
p_train + geom_point(aes(x=YearsExperience, 
                   y=Salary, 
                   color= 'Red')) + 
  geom_line(aes(x=YearsExperience, y=predict(regressor, newdata = training_set)),
            color = 'Blue',
            size = 0.9) +
  xlab("Experience in Years") +
  ylab("Salary") +
  ggtitle("Salary vs Experience")

#Plot the test set 
p_test <- ggplot()
p_test + geom_point(data = test_set,
                    aes(x=test_set$YearsExperience, 
                         y=test_set$Salary, 
                         color= 'Red')) + 
  geom_line(data = training_set,
            aes(x=training_set$YearsExperience, y=predict(regressor, newdata = training_set)),
            color = 'Blue',
            size = 0.9) +
  xlab("Experience in Years") +
  ylab("Salary") +
  ggtitle("Salary vs Experience")

            